modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/ExtensionsNamespaceHandler.java

```diff
@@ -0,0 +1,214 @@
+        }
+
+        return super.parse(element, parserContext);
+    }
+
+    private boolean isHandled(String namespace)
+    {
+        return handledExtensions.containsKey(namespace);
+    }
+
+    private void registerExtensionParsers(String namespace, Element element, ParserContext parserContext)
+    {
+        try
+        {
+            Extension extension = locateExtensionByNamespace(namespace);
+
+            registerTopLevelParameters(extension);
+            registerConfigurations(extension);
+            registerOperations(extension);
+
+            handledExtensions.put(namespace, extension);
+        }
+        catch (Exception e)
+        {
+            parserContext.getReaderContext().fatal(e.getMessage(), element, e);
+        }
+    }
+
+    private void registerOperations(Extension extension) throws Exception
+    {
+        for (Operation operation : extension.getOperations())
+        {
+            registerBeanDefinitionParser(hyphenize(operation.getName()), new OperationBeanDefinitionParser(operation));
+        }
+    }
+
+    private void registerConfigurations(Extension extension) throws Exception
+    {
+        for (Configuration configuration : extension.getConfigurations())
+        {
+            registerBeanDefinitionParser(configuration.getName(), new ConfigurationBeanDefinitionParser(configuration));
+        }
+    }
+
+    private void registerTopLevelParameters(Extension extension)
+    {
+        for (Configuration configuration : extension.getConfigurations())
+        {
+            registerTopLevelParameter(extension, configuration.getParameters());
+        }
+
+        for (Operation operation : extension.getOperations())
+        {
+            registerTopLevelParameter(extension, operation.getParameters());
+        }
+    }
+
+    private void registerTopLevelParameter(final Extension extension, final DataType parameterType)
+    {
+        parameterType.getQualifier().accept(new BaseDataQualifierVisitor()
+        {
+
+            @Override
+            public void onPojo()
+            {
+                String name = hyphenize(getGlobalPojoTypeName(parameterType));
+                if (topLevelParameters.put(extension, name))
+                {
+                    registerBeanDefinitionParser(name, new TopLevelParameterTypeBeanDefinitionParser(parameterType));
+                }
+            }
+
+            @Override
+            public void onList()
+            {
+                if (!ArrayUtils.isEmpty(parameterType.getGenericTypes()))
+                {
+                    registerTopLevelParameter(extension, parameterType.getGenericTypes()[0]);
+                }
+            }
+
+            @Override
+            public void onMap()
+            {
+                DataType[] genericTypes = parameterType.getGenericTypes();
+                if (genericTypes == null)
+                {
+                    return;
+                }
+
+                if (genericTypes.length >= 1)
```

Comment:
Not following this part: do this mean that a Map can have only one generic type in the definition?